src/redux/books/books.js

```diff
@@ -0,0 +1,17 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState: { books: [] },
+  reducers: {
+    addBook: (state, book) => {
+      state.push(book);
+    },
+    removeBook: (state, bookIndex) => {
+      state.splice(bookIndex, 1);
```

Comment:
- [ ] You should define ADD_BOOK and REMOVE_BOOK as action types following the [ducks pattern](https://github.com/erikras/ducks-modular-redux)

Reply:
@EricMbouwe, I thought the whole idea of Redux Toolkit was, among other things, to relieve us of using those constant action types - given the fact that it's returning us actions based on the keys properties of the reducer property of the slice. I also got the idea that switches were no longer needed, if we used the Redux toolkit. If that's still necessary, however, of course, I will implement it.

Reply:
Hi @BucurEva87,

We still need action creators even with redux toolkit. The purpose of that is to follow the the core pattern action => reducer => store => action. You can learn more on this s[section](https://redux-toolkit.js.org/usage/usage-guide) in the documentation